libbeat/common/kubernetes/util_test.go

```diff
@@ -0,0 +1,316 @@
+			podname:     "test-pod",
+			namespace:   "default",
+		},
+		{
+			name:        "test value with inCluster, pod found and env var not set",
+			host:        "",
+			isInCluster: true,
+			node:        "test-node",
+			err:         nil,
+			setEnv:      false,
+			machineid:   "",
+			podname:     "test-pod",
+			namespace:   "default",
+		},
+		{
+			name:        "test value with inCluster, pod found and env var set",
+			host:        "",
+			isInCluster: true,
+			node:        "worker-2",
+			err:         nil,
+			setEnv:      true,
+			machineid:   "",
+			podname:     "test-pod",
+			namespace:   "default",
+		},
+		{
+			name:        "test value without inCluster, machine-id empty and env var not set",
+			host:        "",
+			isInCluster: false,
+			node:        "",
+			err:         ge,
+			setEnv:      false,
+			machineid:   "",
+			podname:     "",
+			namespace:   "",
+		},
+		{
+			name:        "test value without inCluster, machine-id set, node not found and env var not set",
+			host:        "",
+			isInCluster: false,
+			node:        "",
+			err:         ge,
+			setEnv:      false,
+			machineid:   "worker-2",
+			podname:     "",
+			namespace:   "",
+		},
+		{
+			name:        "test value without inCluster, machine-id set, node found and env var not set",
+			host:        "",
+			isInCluster: false,
+			node:        "worker-2",
+			err:         nil,
+			setEnv:      false,
+			machineid:   "worker-2",
+			podname:     "",
+			namespace:   "",
+		},
+		{
+			name:        "test value without inCluster, machine-id set, node not found and env var set",
+			host:        "",
+			isInCluster: false,
+			node:        "worker-2",
+			err:         nil,
+			setEnv:      true,
+			machineid:   "worker-2",
+			podname:     "",
+			namespace:   "",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			if test.setEnv {
+
+				if err := os.Setenv("NODE_NAME", "worker-2"); err != nil {
+					t.Fatal(err)
+				}
+				defer func() {
+					if err := os.Unsetenv("NODE_NAME"); err != nil {
+						t.Fatal(err)
+					}
+				}()
+			}
+			mhu := createMockhu(test.namespace, test.podname, test.machineid)
+
+			if test.name == "test value with inCluster, pod found and env var not set" {
+				err := createPod(client)
+
+				if err != nil {
+					t.Fatal(err)
+				}
+				defer func() {
+					pod := "test-pod"
+					err := client.CoreV1().Pods("default").Delete(context.Background(), pod, metav1.DeleteOptions{})
+					if err != nil {
+						t.Fatalf("failed to delete k8s pod: %v", err)
+					}
+				}()
+			}
+
+			if test.name == "test value without inCluster, machine-id set, node found and env var not set" {
+				err := createNode(client, "worker-2")
+				if err != nil {
+					t.Fatal(err)
+				}
+				defer func() {
+					err := client.CoreV1().Nodes().Delete(context.Background(), "worker-2", metav1.DeleteOptions{})
+					if err != nil {
+						t.Fatalf("failed to delete k8s node: %v", err)
+					}
+				}()
+			}
```

Comment:
This conditional code based on the test name is error prone, a change in the title would unexpectedly change the result of the test.

If some initialization code needs to be run for some test cases, consider adding a `init(*testing.T, kubernetes.client)` function to the test cases struct, so you can do here something like this:
```suggestion
			if test.init != nil {
			    test.init(t, client)
			}
```

Reply:
@MichaelKatsoulis to be clear with my review, this would be the only thing I would like to see changed before merging, the rest of the comments can be seen as suggestions and no blockers at all :slightly_smiling_face:

Reply:
Thanks @jsoriano. Great suggestion! I updated it.